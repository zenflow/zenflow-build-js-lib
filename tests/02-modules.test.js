/* eslint-env jest */

const path = require('path')
const buildFixture = require('./util/buildFixture')
const { readFile, readdir } = require('./util/fs')

describe('modules', () => {
  let tempDir
  beforeAll(async () => {
    tempDir = await buildFixture('modules')
  })
  test('have each file listed in package.json', async () => {
    const expectedFiles = ['cjs', 'es', 'umd'].map(
      format => `modules-fixture.${format}.js`,
    )
    expect(await readdir(path.join(tempDir, 'dist'))).toEqual(expectedFiles)
  })
  test('evaluated module exports', async () => {
    const expected = {
      SQR_SQRT_HALF: 0.5000000000000001,
      SQR_SQRT2: 2.0000000000000004,
    }
    const cjsFile = path.join(tempDir, 'dist/modules-fixture.cjs.js')
    expect(require(cjsFile)).toEqual(expected)
    const umdFile = path.join(tempDir, 'dist/modules-fixture.umd.js')
    expect(require(umdFile)).toEqual(expected)
  })
  test('snapshot', async () => {
    const files = await Promise.all(
      [
        'dist/modules-fixture.cjs.js',
        'dist/modules-fixture.es.js',
        'dist/modules-fixture.umd.js',
      ].map(file => readFile(path.join(tempDir, file))),
    )
    expect(files).toMatchSnapshot()
  })
})
