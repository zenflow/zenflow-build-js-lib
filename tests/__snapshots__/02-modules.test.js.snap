// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`modules snapshot (A) evaluated cjs 1`] = `
Object {
  "SQR_SQRT2": 2.0000000000000004,
  "SQR_SQRT_HALF": 0.5000000000000001,
}
`;

exports[`modules snapshot (A) evaluated umd 1`] = `
Object {
  "SQR_SQRT2": 2.0000000000000004,
  "SQR_SQRT_HALF": 0.5000000000000001,
}
`;

exports[`modules snapshot (B) built cjs 1`] = `
"/** @preserve
  * package: modules-fixture v0.0.0
  * file: dist/modules-fixture.cjs.js
  * homepage: HOMEPAGE
  * license: LICENSE
  **/

'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

function _interopDefault (ex) { return (ex && (typeof ex === 'object') && 'default' in ex) ? ex['default'] : ex; }

var SQRT2 = _interopDefault(require('const-sqrt-two'));

// sqrt(1/2)

var SQRT_HALF = 7.07106781186547524400844362104849039284835937688474036588339868995366239231053519425193767163820786367506923115e-01;

// EXPORTS //

var lib = SQRT_HALF;

var square = (function (number) {
  return number * number;
});

var SQR_SQRT_HALF = square(lib);
var SQR_SQRT2 = square(SQRT2);

exports.SQR_SQRT_HALF = SQR_SQRT_HALF;
exports.SQR_SQRT2 = SQR_SQRT2;
"
`;

exports[`modules snapshot (B) built es 1`] = `
"/** @preserve
  * package: modules-fixture v0.0.0
  * file: dist/modules-fixture.es.js
  * homepage: HOMEPAGE
  * license: LICENSE
  **/

import SQRT2 from 'const-sqrt-two';

// sqrt(1/2)

var SQRT_HALF = 7.07106781186547524400844362104849039284835937688474036588339868995366239231053519425193767163820786367506923115e-01;

// EXPORTS //

var lib = SQRT_HALF;

var square = (function (number) {
  return number * number;
});

var SQR_SQRT_HALF = square(lib);
var SQR_SQRT2 = square(SQRT2);

export { SQR_SQRT_HALF, SQR_SQRT2 };
"
`;

exports[`modules snapshot (B) built umd 1`] = `
"/** @preserve
  * package: modules-fixture v0.0.0
  * file: dist/modules-fixture.umd.js
  * homepage: HOMEPAGE
  * license: LICENSE
  **/

(function (global, factory) {
	typeof exports === 'object' && typeof module !== 'undefined' ? factory(exports, require('const-sqrt-two')) :
	typeof define === 'function' && define.amd ? define(['exports', 'const-sqrt-two'], factory) :
	(factory((global.modulesFixture = {}),global.constSqrtTwo));
}(this, (function (exports,SQRT2) { 'use strict';

	SQRT2 = SQRT2 && SQRT2.hasOwnProperty('default') ? SQRT2['default'] : SQRT2;

	// sqrt(1/2)

	var SQRT_HALF = 7.07106781186547524400844362104849039284835937688474036588339868995366239231053519425193767163820786367506923115e-01;

	// EXPORTS //

	var lib = SQRT_HALF;

	var square = (function (number) {
	  return number * number;
	});

	var SQR_SQRT_HALF = square(lib);
	var SQR_SQRT2 = square(SQRT2);

	exports.SQR_SQRT_HALF = SQR_SQRT_HALF;
	exports.SQR_SQRT2 = SQR_SQRT2;

	Object.defineProperty(exports, '__esModule', { value: true });

})));
"
`;
